integration-tests/tests/semantic_search.rs

```diff
@@ -0,0 +1,164 @@
+                    IngestionDocument {
+                        id: "d3".into(),
+                        snippet: "this is another sentence which we have".into(),
+                        properties: None,
+                    },
+                ],
+            )
+            .await?;
+
+            let SearchResults { documents } = send_assert_json(
+                &client,
+                client
+                    .get(personalization_url.join("/semantic_search/d1")?)
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+
+            if let [first, second] = &documents[..] {
+                assert_eq!(first.id, "d3");
+                assert_eq!(second.id, "d2");
+                assert!(first.score > second.score);
+                assert!(first.properties.is_null());
+                assert_eq!(second.properties, json!({ "dodo": 4 }))
+            } else {
+                panic!("Unexpected number of documents: {documents:?}");
+            }
+
+            Ok(())
+        },
+    )
+    .await;
+}
+
+#[tokio::test]
+async fn test_semantic_search_min_similarity() {
+    test_two_apps::<Ingestion, Personalization, _>(
+        unchanged_config,
+        unchanged_config,
+        |client, ingestion_url, personalization_url, _services| async move {
+            ingest(
+                &client,
+                &ingestion_url,
+                &[
+                    IngestionDocument {
+                        id: "d1".into(),
+                        snippet: "Computers are made of technology.".into(),
+                        properties: None,
+                    },
+                    IngestionDocument {
+                        id: "d2".into(),
+                        snippet: "Mountains smaller then a river.".into(),
+                        properties: None,
+                    },
+                    IngestionDocument {
+                        id: "d3".into(),
+                        snippet: "Computer technology is made".into(),
+                        properties: None,
+                    },
+                ],
+            )
+            .await?;
+
+            let SearchResults { documents } = send_assert_json(
+                &client,
+                client
+                    .get(personalization_url.join("/semantic_search/d1")?)
+                    .query(&[("min_similarity", "0.6")])
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+
+            if let [first] = &documents[..] {
+                assert_eq!(first.id, "d3");
+                assert!(first.score >= 0.6);
+            } else {
+                panic!("Unexpected number of documents: {documents:?}");
+            }
+
+            Ok(())
+        },
+    )
+    .await;
+}
+
+#[derive(Deserialize)]
+struct SearchResults {
+    documents: Vec<PersonalizationDocument>,
+}
+
+#[derive(Debug, Deserialize)]
+struct PersonalizationDocument {
```

Comment:
:art: naming as in the spec
```suggestion
struct PersonalizedDocumentData {
```